src/ffjord.jl

```diff
@@ -61,25 +82,35 @@ function jacobian_fn(f, x::AbstractVector)
    vcat([transpose(back(ȳ(i))[1]) for i = 1:length(y)]...)
  end
 
-function ffjord(du,u,p,t,re,monte_carlo,e)
+function cnf(du,u,p,t,re)
     z = @view u[1:end-1]
     m = re(p)
+    J = jacobian_fn(m, z)
+    trace_jac = length(z) == 1 ? sum(J) : tr(J)
+    du[1:end-1] = m(z)
+    du[end] = -trace_jac
+end
+
+function ffjord(du,u,p,t,re,e,monte_carlo)
+    z = @view u[1:end-3]
+    m = re(p)
+    _, back = Zygote.pullback(m,z)
+    eJ = back(e)[1]
     if monte_carlo
-        _, back = Zygote.pullback(m,z)
-        eJ = back(e)[1]
         trace_jac = (eJ.*e)[1]
     else
         J = jacobian_fn(m, z)
         trace_jac = length(z) == 1 ? sum(J) : tr(J)
     end
-    du[1:end-1] = m(z)
-    du[end] = -trace_jac
+    du[1:end-3] = m(z)
+    du[end-2] = -trace_jac
+    du[end-1] = sum(abs2, m(z))
```

Comment:
so FFJORD would always regularize? Seems like that should be an option.

Reply:
Yeah, in the current implementation it's always regularized. I guess we can put a switch to enable regularization here.